tests/src/test/scala/com/nvidia/spark/rapids/QueryFuzzer.scala

```diff
@@ -0,0 +1,240 @@
+      .set(RapidsConf.INCOMPATIBLE_OPS.key, boolString())
+      .set(RapidsConf.ENABLE_CAST_STRING_TO_FLOAT.key, boolString())
+      .set(RapidsConf.ENABLE_CAST_STRING_TO_DECIMAL.key, boolString())
+      .set(RapidsConf.ENABLE_CAST_STRING_TO_DECIMAL.key, boolString())
+      .set(RapidsConf.ENABLE_CAST_STRING_TO_INTEGER.key, boolString())
+      .set(RapidsConf.ENABLE_CAST_STRING_TO_TIMESTAMP.key, boolString())
+      .set(RapidsConf.ENABLE_CAST_FLOAT_TO_STRING.key, boolString())
+      .set(RapidsConf.ENABLE_CAST_FLOAT_TO_DECIMAL.key, boolString())
+      .set(RapidsConf.ENABLE_CAST_FLOAT_TO_INTEGRAL_TYPES.key, boolString())
+      .set(RapidsConf.ENABLE_CAST_DECIMAL_TO_STRING.key, boolString())
+      .set(RapidsConf.ENABLE_FLOAT_AGG.key, boolString())
+      .set(RapidsConf.ENABLE_HASH_OPTIMIZE_SORT.key, boolString())
+  }
+
+  private def boolString(): String = {
+    rand.nextBoolean().toString
+  }
+
+  private def oneOf(value: String*): String = {
+    value(rand.nextInt(value.length))
+  }
+
+  def randomOperator(spark: SparkSession, depth: Int, maxDepth: Int): DataFrame = {
+
+    // if we have reached maximum depth then create a leaf node
+    if (depth == maxDepth) {
+      return generateDataSource(spark)
+
+      //TODO Add support for reading from CSV and Parquet data sources
+//      if (rand.nextFloat() > 0.7) {
+//        return generateDataSource(spark)
+//      } else {
+//        return renameColumns(
+//          spark.read.parquet("/mnt/tpcds/parquet/sf100-decimals/web_sales.dat"))
+//      }
+    }
+
+    // recurse down first to get a DataFrame
+    val df = randomOperator(spark, depth+1, maxDepth)
+
+    // build a list of available transformations
+    val ctx = FuzzContext(spark, this)
+    val transformations = Seq(
+      Filter(ctx),
+      SortRandomColumns(ctx),
+      RandomCastFromString(ctx),
+      Aggregate(ctx),
+      Repartition(ctx)
+    )
+
+    // filter down to a list of transformations that are applicable in the current context
+    val validTransformations = transformations.filter(_.canTransform(df))
+    if (validTransformations.isEmpty) {
+      df
+    } else {
+      // pick a transformation at random
+      val tf = randomElement(validTransformations)
+      // apply the transformation
+      tf.transform(df)
+    }
+  }
+
+  /**
+   * Generate a random in-memory DataFrame.
+   */
+  def generateDataSource(spark: SparkSession) = {
+    val numFields = 1 + rand.nextInt(32)
+    val supportedTypes = Seq(DataTypes.IntegerType,
+      DataTypes.StringType,
+      new DecimalType(10,2),
+      DataTypes.DateType,
+      DataTypes.TimestampType,
+      DataTypes.BooleanType,
+      DataTypes.DoubleType,
+      DataTypes.FloatType)
+    val dataTypes = (0 until numFields).map(_ => randomElement(supportedTypes))
+    val schema = FuzzerUtils.createSchema(dataTypes)
+    val options = FuzzerOptions(validStringChars = Some(" \t\r\n0123456789.+-/:aidfnT"),
```

Comment:
Curious why this tests such a limited character set -- no multi-byte Unicode characters for example.